### PR TITLE
Add Runtime Option to Ignore Air in Material Plots

### PR DESCRIPTION
**Pull request by klendathu2k**
Implement an option so that Air is only scored at the CAVE and HALL levels of the geometries.

When an envelope volume (aka assembly volume) contains significant empty space and/or very lightweight daughters, the air in the envelope can contribute a relatively large amount to the material budget plots.  This commit adds an option so that users can ignore air w/in the material budget calculations (and get at only the physical detector itself).  

Air will still count towards the HALL and CAVE histograms when this option is used.

This option is off by default.

### CODE REVIEW

File: StarVMC/StarAgmlChecker/StarAgmlChecker.cxx [hunk 1]
```diff
@@ -151,7 +141,21 @@ void StarAgmlChecker::Fill( TObjectSet *set, Double_t rmin, Double_t rmax, Doubl
...
+	  matprop = material->GetRadLen();
+	  TString nm = material->GetName();
+	  nm.ToLower();
+	  if ( nm.Contains("air") ) {                                   // use the name of the material to select air...
+	    if ( false == mScoreAir ) {                                 // user option to ignore air in detector volumes
+	      if ( !name.Contains("CAVE") && !name.Contains("HALL") ) { // will still count air in the cave and hall histograms
+		matprop = 0;                                            // property will not be scored
+	      }
+	    }
+	  }
```
Comment (plexoos) [inline StarVMC/StarAgmlChecker/StarAgmlChecker.cxx:155]: For better readability please consider replacing the nested 'if' statements with a single one such as:

```c++
bool condition = conditionA && !conditionB && conditionC && !conditionD;
matprop = 0;
if (condition) {
  matprop = material->GetRadLen();
}
```

File: StarVMC/StarAgmlChecker/StarAgmlChecker.h [hunk 2]
```diff
@@ -107,6 +107,8 @@ class StarAgmlChecker : public TGeoChecker
 			    const Double_t  zmax  = +4000.0,
 			    const Option_t *opts = "top" );
 
+  
+
```
Comment (plexoos) [inline StarVMC/StarAgmlChecker/StarAgmlChecker.h:111]: Remove if not intentional
```suggestion
```

**Review by plexoos — APPROVED** [review]
The code appears to deliver what is promised in the description

**Review by jdbrice — APPROVED** [review]
Looks good. I'll approve with the expectation that @plexoos 's comment is addressed

**Pull request merged**